google/cloud/bigquery/dbapi/cursor.py

```diff
@@ -212,13 +217,74 @@ def _try_fetch(self, size=None):
 
         if self._query_data is None:
             client = self.connection._client
+            bqstorage_client = self.connection._bqstorage_client
+
+            if bqstorage_client:
+                try:
+                    rows_iterable = self._bqstorage_fetch(bqstorage_client)
+                    self._query_data = _helpers.to_bq_table_rows(rows_iterable)
+                    return
+                except google.api_core.exceptions.Forbidden:
+                    # Don't hide errors such as insufficient permissions to create
+                    # a read session, or the API is not enabled. Both of those are
+                    # clearly problems if the developer has explicitly asked for
+                    # BigQuery Storage API support.
+                    raise
```

Comment:
Wgt catch an exception only to re-raise it without any further processing here? It seems likely to create a confusing traceback.

Reply:
This was modeled after a [similar logic](https://github.com/googleapis/google-cloud-python/blob/b387134827dbc3be0e1b431201e0875798002fda/bigquery/google/cloud/bigquery/table.py#L1407-L1436) in `table.RowItrator`.

I _think_ the intent was to explain why a particular subset of exceptions (Forbidden) does not result in a fallback to the default client, but a reraise instead, although that catch-reraise block is indeed redundant. I'll remove it, and only keep the comment.